### PR TITLE
[2.x] Merge sbt-projectmatrix

### LINKED ISSUE #7641: [2.x] In-source projectMatrix
https://github.com/sbt/sbt-projectmatrix

### PR DESCRIPTION
**Pull request by adpi2**
Fixes #7641

### CODE REVIEW

File: main/src/main/scala/sbt/ProjectMatrix.scala [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+                  List(scalaVersion := sv)
+                case _ =>
+                  List(autoScalaLibrary := false, crossPaths := false)
+              }
+            )
+            .settings(
+              outputPath := {
+                val o = outputPath.value
+                if nonScalaNorPlatformDirSuffix.nonEmpty then s"$o/$nonScalaNorPlatformDirSuffix"
+                else o
```
Comment (adpi2) [inline main/src/main/scala/sbt/ProjectMatrix.scala:320]: No need a custom path for platform anymore, it's in by default.

Comment (adpi2) [inline main/src/main/scala/sbt/ProjectMatrix.scala:320]: No need a custom path for platform anymore, it's already in the path by default.

File: sbt-app/src/sbt-test/project-matrix/js/pending
Comment (adpi2) [inline sbt-app/src/sbt-test/project-matrix/js/pending:1]: Js and Native tests are pending

File: main/src/main/scala/sbt/ProjectMatrix.scala [hunk 2]
```diff
@@ -0,0 +1,818 @@
...
+            }
+            .headOption
+            .getOrElse(sys.error(s"platform axis is missing in $axes"))
+          val childId = projectIds(r)
+          val deps = dependencies.map { resolveMatrixDependency(_, r) } ++ nonMatrixDependencies
+          val aggs = aggregate.map { case ref: LocalProjectMatrix =>
+            val other = lookupMatrix(ref)
+            resolveMatrixAggregate(other, r)
+          } ++ nonMatrixAggregate
+          val dotSbtMatrix = new java.io.File(".sbt") / "matrix"
```
Comment (adpi2) [inline main/src/main/scala/sbt/ProjectMatrix.scala:298]: I wonder if we should change that path to `target/matrix`.

**Comment by eed3si9n** [comment]
Could we keep the git history from the original plz? It's only around 100 commits.
https://eed3si9n.com/splitting-git-repo/ has some related git commands.

**Comment by adpi2** [comment]
> Could we keep the git history from the original plz? It's only around 100 commits.

Here we are

File: main/src/main/scala/sbt/ProjectMatrix.scala (hunk 2 above)
Comment (eed3si9n) [reply main/src/main/scala/sbt/ProjectMatrix.scala:298]: Yea, something we can explore post merge I guess. Not sure how IDEs would behave if the base directories get wiped up.

**Review by eed3si9n — APPROVED** [review]
Thanks!

**Pull request merged**